src/kernel/src/rtld/mod.rs

```diff
@@ -792,6 +792,17 @@ impl<E: ExecutionEngine> RuntimeLinker<E> {
 
                     (Relocated::Tls((md, index)), value)
                 }
+                Relocation::R_X86_64_DTPOFF64 => {
+                    let md = match resolver.resolve_with_local(md, sym, symflags) {
+                        Some((md, _)) => md,
+                        None => continue,
+                    };
+
+                    let sym = md.symbol(sym).unwrap();
+                    let addr = sym.value().wrapping_add_signed(addend);
+
+                    (Relocated::Data((md, addr)), addr)
+                }
```

Comment:
What I saw is it also read the value from the target address and add some values to it.

Reply:
data:=(data + rela^.r_addend + def^.st_value);